kedro/contrib/idempotent/idempotent_state_storage.py

```diff
@@ -25,13 +29,27 @@ def __init__(self, run_id_state=None, input_state=None, nodes_have_been_run=None
         self.nodes_have_been_run = nodes_have_been_run
 
     @staticmethod
-    def generate_run_id():
-        return str(uuid4())
+    def generate_run_id(data_name: str = None, data: Any = None, data_set_type: Any = None):
+        run_id = str(uuid4())
+        if data_name is None:
+            return run_id
+        if data_name.startswith("params:") or data_set_type is MemoryDataSet:
```

Comment:
You only need to optionally pass in `data`.

`def generate_run_id(data=None)`

Then use the original `MemoryDataSet` check, and call the appropriate `update_run_id`

```
if type(catalog._data_sets[name]) is MemoryDataSet:	
        state.update_run_id(name, catalog.load(name))
else:
        state.update_run_id(name)
```

Obviously, change the `update_run_id` signature to match.